test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,644 @@
+        insertAndGetFromKVTable(newScope, newKVT, keyFamily);
+
+        // Cleans up stream resources by sealing existing stream segments & then deletes the stream.
+        cleanUpStream(newScope, streamName);
+        // deleting the scope.
+        cleanUpScope(newScope);
+    }
+
+    /**
+     * Tests Creation and deletion of KVTables and Streams +
+     * under different scopes
+     */
+    @Test
+    public void testDeleteKVTableStream() throws ExecutionException, InterruptedException {
+        String newScope1 = "TestScope1-" + randomAlphanumeric(5);
+        String newScope2 = "TestScope2-" + randomAlphanumeric(5);
+        KeyValueTableInfo newKVT1 = newKeyValueTableName(newScope1);
+        KeyValueTableInfo newKVT2 = newKeyValueTableName(newScope2);
+        KeyValueTableConfiguration config = KeyValueTableConfiguration.builder().partitionCount(5).build();
+        String streamName1 = newKVT1.getKeyValueTableName();
+        String streamName2 = newKVT2.getKeyValueTableName();
+        String readerGroupName = "readerGroup";
+        String readerName = "reader";
+        String routingKey = "routingKey";
+        String streamEvent = "Stream event";
+        int numEvents = 100;
+
+        assertTrue("The requested Scope must get created", this.controller.createScope(newScope1).join());
+        // Creating KVT1 under Scope1.
+        assertTrue("The requested KVT must get created",
+                this.controller.createKeyValueTable(newScope1, newKVT1.getKeyValueTableName(),
+                        config).join());
+
+        assertEquals("Table partition value should be same as passed", DEFAULT_CONFIG.getPartitionCount(),
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegmentsForKeyValueTable(newScope1,
+                        newKVT1.getKeyValueTableName()), RuntimeException::new).getSegments().size());
+
+        val tableSegments1 = this.controller.getCurrentSegmentsForKeyValueTable(newScope1,
+                newKVT1.getKeyValueTableName()).join();
+
+        for (val s : tableSegments1.getSegments()) {
+            // We know there's nothing in these segments. But if the segments hadn't been created, then this will throw
+            // an exception.
+            this.tableStore.get(s.getKVTScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT).join();
+        }
+
+        // Case-1: Deleting KeyValueTable where Scope does not exist.
+        ResultObserver<DeleteKVTableStatus> result1 = new ResultObserver<>();
+        this.controllerService.deleteKeyValueTable(ModelHelper.createKeyValueTableInfo(newScope2,
+                newKVT1.getKeyValueTableName()), result1);
+        assertEquals("The requested table to delete must not be found for invalid Scope name",
+                DeleteKVTableStatus.Status.TABLE_NOT_FOUND, result1.get().getStatus());
+
+        // Case-2: Delete KeyValueTable where KVTable does not exist within the Scope.
```

Comment:
Different test cases should go into different test methods, so that the failure in one doesn't mask failure in others and each test verifies largely a single behavior, among other things. 

A good mental model of distributing tests according to the single responsibility principle is: `UnitOfWork_StateUnderTest_ExpectedBehavior` (even though we don't really use [Roy Oshreve](https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html)'s naming convention in Pravega). 

IMO this test case should be broken down into these smaller test cases:
* deleteTableReturnsTableNotFoundWhenTableDoesn'tExist
* deleteTableReturnsTableNotFoundWhenScopeDoesn'tExist

Also, it makes it much easier for others to understand the test when its name specified not only the unit of work (`deleteTable`), but also the `state under test` and `expected outcome`. Reasonably long names for tests (unlike in main classes) is not considered bad, as long as they add value.